### ISSUE TITLE
msfdb [re]init fails - Error creating MSF web service user API token

### ISSUE THREAD

**Issue by cohall-2020** [rapid7/metasploit-framework#14519]
#### Issue description

Error creating MSF web service user API token

#### Steps to reproduce the issue

1.  msfdb [re]init
2. Enter a username & password

#### What's the expected result?

- msf web service starts

#### What's the actual result?

- msf web service fails to start

#### Platform

- MacOS 11.1 Big Sur/x64

#### Version

- 6.0.22-dev-1a8911e934ca61355c6193c29e1071c69217aad6

#### Additional details

- Output:
```
[?] Would you like to delete your existing data and configurations?: y
====================================================================
Running the 'reinit' command for the database:
Stopping database at /Users/msfuser/.msf4/db
Deleting all data at /Users/msfuser/.msf4/db
Creating database at /Users/msfuser/.msf4/db
Starting database at /Users/msfuser/.msf4/db...success
Creating database users
Writing client authentication configuration file /Users/msfuser/.msf4/db/pg_hba.conf
Stopping database at /Users/msfuser/.msf4/db
Starting database at /Users/msfuser/.msf4/db...success
Creating initial database schema
====================================================================

====================================================================
Running the 'reinit' command for the webservice:
Stopping MSF web service PID xxxxx
[?] Initial MSF web service account username? [msfuser]: msfuser
[?] Initial MSF web service account password? (Leave blank for random password):
Generating SSL key and certificate for MSF web service
Attempting to start MSF web service...success
MSF web service started and online
Creating MSF web service user msfuser

    ############################################################
    ##              MSF Web Service Credentials               ##
    ##                                                        ##
    ##        Please store these credentials securely.        ##
    ##    You will need them to connect to the webservice.    ##
    ############################################################

MSF web service username: msfuser
MSF web service password: [...]
[!] Error creating MSF web service user API token
Failed to complete MSF web service configuration, please reinitialize.
Stopping MSF web service PID xxxxx
====================================================================
```

- log info:
```
2020-12-15 13:53:30 - NoMethodError - undefined method `api_unauthenticated_path' for #<Msf::WebServices::MetasploitApiApp:0x0000000109806f10>:
        /opt/metasploit-framework/embedded/framework/lib/msf/core/web_services/servlet/auth_servlet.rb:82:in `block in post_generate_token'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1675:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1675:in `block in compile!'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1013:in `block (3 levels) in route!'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1032:in `route_eval'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1013:in `block (2 levels) in route!'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1061:in `block in process_route'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1059:in `catch'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1059:in `process_route'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1011:in `block in route!'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1008:in `each'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1008:in `route!'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1129:in `block in dispatch!'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `block in invoke'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `catch'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `invoke'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1124:in `dispatch!'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:939:in `block in call!'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `block in invoke'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `catch'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `invoke'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:939:in `call!'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:929:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/warden-1.2.9/lib/warden/manager.rb:36:in `block in call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/warden-1.2.9/lib/warden/manager.rb:34:in `catch'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/warden-1.2.9/lib/warden/manager.rb:34:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/xss_header.rb:18:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/base.rb:50:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/base.rb:50:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/path_traversal.rb:16:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/json_csrf.rb:26:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/base.rb:50:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/base.rb:50:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rack-protection-2.1.0/lib/rack/protection/frame_options.rb:31:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/session/abstract/id.rb:266:in `context'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/session/abstract/id.rb:260:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/null_logger.rb:11:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/head.rb:12:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:216:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1991:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1542:in `block in call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1769:in `synchronize'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1542:in `call'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/thin-1.8.0/lib/thin/connection.rb:86:in `block in pre_process'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/thin-1.8.0/lib/thin/connection.rb:84:in `catch'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/thin-1.8.0/lib/thin/connection.rb:84:in `pre_process'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/thin-1.8.0/lib/thin/connection.rb:53:in `process'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/thin-1.8.0/lib/thin/connection.rb:39:in `receive_data'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/eventmachine-1.2.7/lib/eventmachine.rb:195:in `run_machine'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/eventmachine-1.2.7/lib/eventmachine.rb:195:in `run'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/thin-1.8.0/lib/thin/backends/base.rb:75:in `start'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/thin-1.8.0/lib/thin/server.rb:162:in `start'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/thin-1.8.0/lib/thin/controllers/controller.rb:87:in `start'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/thin-1.8.0/lib/thin/runner.rb:203:in `run_command'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/thin-1.8.0/lib/thin/runner.rb:159:in `run!'
        /opt/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/thin-1.8.0/bin/thin:6:in `<top (required)>'
        /opt/metasploit-framework/embedded/bin/thin:23:in `load'
        /opt/metasploit-framework/embedded/bin/thin:23:in `<main>'
2020-12-15 13:53:32 -0500 Exiting!
```

**Comment by Spl01ter** [comment]
$ msfdb reinit
[?] Would you like to delete your existing data and configurations?: yes
====================================================================
Running the 'reinit' command for the database:
Stopping database at /Users/ac/.msf4/db
Deleting all data at /Users/ac/.msf4/db
Creating database at /Users/ac/.msf4/db
Starting database at /Users/user/.msf4/db...success
Creating database users
Writing client authentication configuration file /Users/user/.msf4/db/pg_hba.conf
Stopping database at /Users/user/.msf4/db
Starting database at /Users/user/.msf4/db...success
Creating initial database schema
====================================================================

====================================================================
Running the 'reinit' command for the webservice:
Stopping MSF web service PID 15563
[?] Initial MSF web service account username? [xxx]: 
[?] Initial MSF web service account password? (Leave blank for random password): 
Generating SSL key and certificate for MSF web service
Attempting to start MSF web service...success
MSF web service started and online
Creating MSF web service user ac

    ############################################################
    ##              MSF Web Service Credentials               ##
    ##                                                        ##
    ##        Please store these credentials securely.        ##
    ##    You will need them to connect to the webservice.    ##
    ############################################################

MSF web service username: xxx
MSF web service password: xxxxxxx
[!] Error creating MSF web service user API token
Failed to complete MSF web service configuration, please reinitialize.
Stopping MSF web service PID 15650
====================================================================

**Comment by Spl01ter** [comment]
$ msfdb start
====================================================================
Running the 'start' command for the database:
Starting database at /Users/ac/.msf4/db...success
====================================================================

====================================================================
Running the 'start' command for the webservice:
Attempting to start MSF web service...success
MSF web service started and online
====================================================================

**Comment by Spl01ter** [comment]
[-] Error while running command db_connect: Failed to connect to the HTTP data service: Data Service does not appear to be responding

Call stack:
/opt/metasploit-framework/embedded/framework/lib/msf/ui/console/command_dispatcher/db.rb:2067:in `rescue in db_connect_http'
/opt/metasploit-framework/embedded/framework/lib/msf/ui/console/command_dispatcher/db.rb:2062:in `db_connect_http'
/opt/metasploit-framework/embedded/framework/lib/msf/ui/console/command_dispatcher/db.rb:1828:in `cmd_db_connect'
/opt/metasploit-framework/embedded/framework/lib/rex/ui/text/dispatcher_shell.rb:525:in `run_command'
/opt/metasploit-framework/embedded/framework/lib/rex/ui/text/dispatcher_shell.rb:476:in `block in run_single'
/opt/metasploit-framework/embedded/framework/lib/rex/ui/text/dispatcher_shell.rb:470:in `each'
/opt/metasploit-framework/embedded/framework/lib/rex/ui/text/dispatcher_shell.rb:470:in `run_single'
/opt/metasploit-framework/embedded/framework/lib/msf/ui/console/driver.rb:255:in `block in load_db_config'
/opt/metasploit-framework/embedded/framework/lib/msf/ui/console/driver.rb:242:in `each_pair'
/opt/metasploit-framework/embedded/framework/lib/msf/ui/console/driver.rb:242:in `load_db_config'
/opt/metasploit-framework/embedded/framework/lib/msf/ui/console/driver.rb:132:in `initialize'
/opt/metasploit-framework/embedded/framework/lib/metasploit/framework/command/console.rb:62:in `new'
/opt/metasploit-framework/embedded/framework/lib/metasploit/framework/command/console.rb:62:in `driver'
/opt/metasploit-framework/embedded/framework/lib/metasploit/framework/command/console.rb:48:in `start'
/opt/metasploit-framework/embedded/framework/lib/metasploit/framework/command/base.rb:82:in `start'
/opt/metasploit-framework/bin/../embedded/framework/msfconsole:23:in `<main>'

**Comment by Spl01ter** [comment]
Same problem here

**Comment by adfoster-r7** [comment]
Thanks for the error report, this should be fixed in the next release of Metasploit
https://github.com/rapid7/metasploit-framework/pull/14499

**Comment by adfoster-r7** [comment]
This should be fixed in release 6.0.23
